express-notes-api/index.js

```diff
@@ -0,0 +1,138 @@
+const express = require('express');
+const app = express();
+const data = require('./data.json');
+const fs = require('fs');
+
+app.get('/api/notes', function (req, res) {
+  const notesArr = [];
+  for (const note in data.notes) {
+    notesArr.push(data.notes[note]);
+  }
+  res.status(200);
+  res.json(notesArr);
+});
+
+app.get('/api/notes/:id', function (req, res) {
+  const id = req.params.id;
+  if (!(id > 0)) {
+    const err = {
+      error: 'id must be a positive integer'
+    };
+    res.status(400);
+    res.json(err);
+  }
+  if (!Object.prototype.hasOwnProperty.call(data.notes, id)) {
+    const err = {
+      error: `cannot find note with id ${id}`
+    };
+    res.status(404);
+    res.json(err);
+  }
+  const note = data.notes[id];
+  res.status(200);
+  res.json(note);
```

Comment:
Be sure to include `return` guards or `else` blocks. `res.send()` and `res.json()` do not halt the rest of the JS code from executing and you don't want your server accidentally attempting to response more than once to the same request (it'll crash).